zipp/__init__.py

```diff
@@ -313,6 +322,47 @@ def iterdir(self):
         subs = map(self._next, self.root.namelist())
         return filter(self._is_child, subs)
 
+    def match(self, path_pattern):
+        return pathlib.Path(self.at).match(path_pattern)
+
+    def is_symlink(self):
+        return False  # See #82102
+
+    @contextlib.contextmanager
+    def _scandir(self):  # Needed for glob
+        yield self.iterdir()
+
+    def _make_child_relpath(self, relpath):  # Needed for glob
+        return self / relpath
+
+    def glob(self, pattern):
+        if not pattern:
+            raise ValueError("Unacceptable pattern: {!r}".format(pattern))
+        path = pathlib.Path(self.at)
+        drv, root, pattern_parts = path._flavour.parse_parts((pattern,))
+        if drv or root:
+            raise NotImplementedError("Non-relative patterns are unsupported")
+        if pattern[-1] in (path._flavour.sep, path._flavour.altsep):
+            pattern_parts.append('')
+        selector = pathlib._make_selector(tuple(pattern_parts), path._flavour)
+        for p in selector.select_from(self):
+            yield p
```

Comment:
I'm beginning to think that copying the behavior from pathlib.Path isn't the best option. I'm slightly surprised that `_flavour` is relevant here, because zip files use forward slashes and it's probably undesirable to support backslashes on Windows only (I'm on a mission to reduce the amount of divergence across platforms). I'd like to consider re-implementing this behavior with tighter constraints on what's supported and avoid concerns about flavour.

I'll plan to explore this concern later.

Reply:
In e587f319826bce03df1dc7233842f054172c1fd2, I re-implemented the glob behavior more simply and without using private interfaces of the pathlib library.